### PR TITLE
Search page for shapes and ontologies

### PR DESCRIPTION
**Pull request by Huber-David**

### CODE REVIEW

File: public/locales/de/translation.json [hunk 1]
```diff
@@ -19,10 +19,11 @@
     "about": "Über",
     "support": "Unterstützung",
     "english": "English",
-    "spanish": "Spanisch",
+    "spanish": "Español",
```
Comment (Novak-Zsolt) [inline public/locales/de/translation.json:22]: We removed the Spanish language from the Portal. You can delete these translation properties. Sorry for leaving them in :)

File: public/locales/en/translation.json [hunk 2]
```diff
@@ -19,10 +19,11 @@
     "about": "About",
     "support": "Support",
     "english": "English",
-    "spanish": "Spanish",
-    "german": "German",
+    "spanish": "Español",
```
Comment (Novak-Zsolt) [inline public/locales/en/translation.json:22]: As I mentioned previously, we can delete this property.

File: public/locales/en/translation.json [hunk 3]
```diff
@@ -19,10 +19,11 @@
...
+    "german": "Deutsch",
```
Comment (Novak-Zsolt) [inline public/locales/en/translation.json:23]: This being the EN Translation file, shouldn't we leave the Deutsch as German, it's English definition?

File: src/App.js [hunk 4]
```diff
@@ -29,6 +29,7 @@ import {Column} from './common/styles';
 import {Footer} from "./components/footer/Footer";
 
 import './App.css';
+import ShapesAndOntologies from "./components/shapesAndOntologies/ShapesAndOntologies";
```
Comment (Novak-Zsolt) [inline src/App.js:32]: Just a small preference, but please organise imports as following:
1st are the imports regarding external npm packages, default imports come first.
2nd are the imports of our written components or resources.
3nd are the style related imports but this should be only the CSS modules.

**Comment by Huber-David** [comment]
Added serachbar to resource page

File: src/components/resources/Resources.tsx [hunk 5]
```diff
@@ -12,19 +12,22 @@ import Title from "components/Title/Title";
 import Filter from "components/filter/Filter";
 import Text from "components/Text/Text";
 import { useFilters } from "context/ResourceFilterContext";
-import { useFilter } from "hooks/useFilter";
 import { useTranslation } from "react-i18next";
+import { useResourceFilter } from "hooks/useResourceFilter";
 
 import styles from "./Resources.module.css";
+import SearchBar from "../searchBar/SearchBar";
```
Comment (Novak-Zsolt) [inline src/components/resources/Resources.tsx:19]: Again, this is personal preference but let's have this import in the second category of imports, and not below the CSS module.

File: src/components/shapesAndOntologies/ShapesAndOntologies.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+import { useTranslation } from "react-i18next";
+import { useContext, useEffect, useState } from "react";
+import { AuthContext } from "../../context/AuthContextProvider";
+import car from "../../assets/car.gif";
+import { ApiService } from "../../services/ApiService";
+import { mapOntologies, mapShapesAndOntologies, Ontology } from "../../utils/dataMapper";
+import SelfDescriptionCard from "../cards/SelfDescriptionCard";
+import { AuthContextValues } from "../../context/AuthContextValues";
+import Text from "../Text/Text";
+import SearchBar from "../searchBar/SearchBar";
```
Comment (Novak-Zsolt) [inline src/components/shapesAndOntologies/ShapesAndOntologies.tsx:12]: Let's reorganize the imports here as well to maintain a similarity between components.

File: src/components/shapesAndOntologies/ShapesAndOntologies.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+const ShapesAndOntologies = () => {
+    const { t } = useTranslation();
+    const authContext = useContext(AuthContext);
+    const [isLoading, setIsLoading] = useState(false);
+    const [originalOntologies, setOriginalOntologies] = useState<Ontology[]>([]);
+    const [filteredOntologies, setFilteredOntologies] = useState<Ontology[]>([]);
+
+    // todo im not completely satisfied with the way i fetch the ontologies. Would it be better to have the fetching logic in a separate file?
+    const fetchOntologies = async (authContext: AuthContextValues) => {
```
Comment (Novak-Zsolt) [inline src/components/shapesAndOntologies/ShapesAndOntologies.tsx:22]: Yes, let's extract this logic in a custom hook, then import that and use the referenced functions.

File: src/components/filter/Filter.tsx [hunk 8]
```diff
@@ -1,22 +1,25 @@
-import React from "react";
-
+import { FC } from "react";
 import Title from "components/Title/Title";
-import { useFilter } from "hooks/useFilter";
-import { useFilters } from "context/ResourceFilterContext";
-
 import styles from "./Filter.module.css";
 import { FilterSection } from "./FilterSection";
+import { Asset } from "../../hooks/filterAssets";
```
Comment (Novak-Zsolt) [inline src/components/filter/Filter.tsx:5]: Filter section and Asset should be above the CSS module import.

File: public/locales/en/translation.json (hunk 3 above)
Comment (Huber-David) [reply public/locales/en/translation.json:24]: I think its more user friendly if its in the same Language the user is looking for. For an more extreme example, imagine visiting an chinese website and looking for the english button but you dont know what english is in chinese. What do you think?

File: src/App.js (hunk 4 above)
Comment (Huber-David) [reply src/App.js]: I updated all imports and add the rules in eslint

File: src/components/shapesAndOntologies/ShapesAndOntologies.tsx (hunk 7 above)
Comment (Huber-David) [reply src/components/shapesAndOntologies/ShapesAndOntologies.tsx]: I extracted the logic to datamapper

File: src/hooks/useResourceFilter.tsx [hunk 9]
```diff
@@ -1,68 +1,27 @@
...
-    {
-      checkboxName: "scenario",
-      label: "Scenario",
-    },
-    {
-      checkboxName: "environmentModel",
-      label: "Environment Model",
-    },
-  ];
+import { AuthContext } from '../context/AuthContextProvider'; // Update this path as needed
```
Comment (Novak-Zsolt) [inline src/hooks/useResourceFilter.tsx:4]: This comment is not necessary here :)

**Review by Novak-Zsolt — APPROVED** [review]

**Pull request merged**